File: backend/app/models/timetable_entry.py
from sqlalchemy import Column, Integer, ForeignKey, UniqueConstraint
from app.database import Base

class TimetableEntry(Base):
    __tablename__ = "timetable_entries"

    id = Column(Integer, primary_key=True, index=True)
    timetable_id = Column(Integer, ForeignKey("timetables.id", ondelete="CASCADE"))
    class_id = Column(Integer, ForeignKey("classes.id", ondelete="CASCADE"))
    subject_id = Column(Integer, ForeignKey("subjects.id", ondelete="CASCADE"))
    teacher_id = Column(Integer, ForeignKey("teachers.id", ondelete="CASCADE"))
    room_id = Column(Integer, ForeignKey("rooms.id", ondelete="CASCADE"))
    day_id = Column(Integer, ForeignKey("days.id", ondelete="CASCADE"))
    time_slot_id = Column(Integer, ForeignKey("time_slots.id", ondelete="CASCADE"))

    __table_args__ = (
        UniqueConstraint("teacher_id", "day_id", "time_slot_id", name="uq_teacher_slot"),
        UniqueConstraint("room_id", "day_id", "time_slot_id", name="uq_room_slot"),
        UniqueConstraint("class_id", "day_id", "time_slot_id", name="uq_class_slot"),
    )

File: SmartTimetable/src/SmartTimetable.Core/Entities/TimetableEntry.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace SmartTimetable.Core.Entities;

[Table("timetable_entries")]
public class TimetableEntry
{
    [Key]
    [Column("id")]
    public int Id { get; set; }

    [Column("timetable_id")]
    public int? TimetableId { get; set; }

    [Column("class_id")]
    public int? ClassId { get; set; }

    [Column("subject_id")]
    public int? SubjectId { get; set; }

    [Column("teacher_id")]
    public int? TeacherId { get; set; }

    [Column("room_id")]
    public int? RoomId { get; set; }

    [Column("day_id")]
    public int? DayId { get; set; }

    [Column("time_slot_id")]
    public int? TimeSlotId { get; set; }

    // Navigation properties
    [ForeignKey("TimetableId")]
    public Timetable? Timetable { get; set; }

    [ForeignKey("ClassId")]
    public Class? Class { get; set; }

    [ForeignKey("SubjectId")]
    public Subject? Subject { get; set; }

    [ForeignKey("TeacherId")]
    public Teacher? Teacher { get; set; }

    [ForeignKey("RoomId")]
    public Room? Room { get; set; }

    [ForeignKey("DayId")]
    public Day? Day { get; set; }

    [ForeignKey("TimeSlotId")]
    public TimeSlot? TimeSlot { get; set; }
}
